File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
var makeEqual = function(words) {
      let totalCharacters = 0;
    let mp = new Map();

    for (let str of words) {
        for (let ch of str) {
            totalCharacters++;
            mp.set(ch, (mp.get(ch) || 0) + 1);
        }
    }

    if (totalCharacters % words.length !== 0) {
        return false;
    }

    for (let [key, value] of mp) {
        if (value % words.length !== 0) {
            return false;
        }
    }

    return true;
};

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        int TotalCharacters=0;
        unordered_map<char,int>mp;

        for(auto str:words){
            for(auto ch:str){
                TotalCharacters++;
                mp[ch]++;
            }
        }

        if(TotalCharacters%(int)words.size()!=0) return false; 

        for(auto x:mp){
            if(x.second%(int)words.size()!=0)
            return false;
        }

        return true;
    }
};